hello-libs/app/build.gradle

```diff
@@ -42,10 +41,18 @@ android {
 
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
-    compile 'com.android.support:appcompat-v7:25.2.0'
-    compile 'com.android.support.constraint:constraint-layout:1.0.1'
+    compile 'com.android.support:appcompat-v7:26.1.0'
+    compile 'com.android.support.constraint:constraint-layout:1.0.2'
     // uncomment out this one to generate lib binaries,
     // and also uncommented out the one in settings.gradle
     // after lib is generated, just comment them out again
 //        compile project(':gen-libs')
 }
+
+tasks.whenTaskAdded { task ->
+    if (task.name == 'externalNativeBuildRelease') {
+        task.dependsOn ":gen-libs:externalNativeBuildRelease"
+    } else if (task.name == 'externalNativeBuildDebug') {
```

Comment:
thanks for the patch!  may you add:

- a line of comment to indicate that it is for building the lib only
- by default, I think we still want to build just the application, do not want to build the lib. maybe put the whole lib build into comment ( I will also update the readme.md to reflect it ).

To build app and lib ( and link at different location ) has some duplicate work: for shared libs, when building lib, it is packed to APK ( relating to jniLibs of AAR ); then when linking with app's  CMakeLists.txt ( import clause for shared lib), it will be packed again; the second time probably just overwrite the first version; not important, like to avoid pointing people to the wrong direction. kindly let me know what you think, thanks!